src/docker/build.rs

```diff
@@ -0,0 +1,125 @@
+use std::env;
+use std::process::Command;
+use std::str::FromStr;
+
+use super::engine::Engine;
+use crate::errors::*;
+use crate::shell::Verbosity;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Progress {
+    Plain,
+    Auto,
+    Tty,
+}
+
+impl FromStr for Progress {
+    type Err = eyre::ErrReport;
+
+    fn from_str(progress: &str) -> Result<Self> {
+        Ok(match progress {
+            "plain" => Progress::Plain,
+            "auto" => Progress::Auto,
+            "tty" => Progress::Tty,
+            s => eyre::bail!("unexpect progress type: expected plain, auto, or tty and got {s}"),
+        })
+    }
+}
+
+impl From<Progress> for &str {
+    fn from(progress: Progress) -> Self {
+        match progress {
+            Progress::Plain => "plain",
+            Progress::Auto => "auto",
+            Progress::Tty => "tty",
+        }
+    }
+}
+
+pub trait BuildCommandExt {
+    fn build(&mut self) -> &mut Self;
+    fn progress(&mut self, progress: Option<Progress>) -> Result<&mut Self>;
+    fn verbose(&mut self, verbosity: Verbosity) -> &mut Self;
+    fn scan_suggest(&mut self, suggest: bool) -> &mut Self;
+    fn cross_labels(&mut self, target: &str, platform: &str) -> &mut Self;
+}
+
+impl BuildCommandExt for Command {
+    fn build(&mut self) -> &mut Self {
+        match Engine::has_buildkit() {
+            true => self.args(["buildx", "build"]),
+            false => self.arg("build"),
+        }
+    }
+
+    fn progress(&mut self, progress: Option<Progress>) -> Result<&mut Self> {
+        let progress: Progress = match progress {
+            None => env::var("CROSS_BUILD_PROGRESS")
+                .as_deref()
+                .unwrap_or("auto")
+                .parse()?,
+            Some(progress) => progress,
+        };
+        Ok(self.args(["--progress", progress.into()]))
+    }
+
+    fn verbose(&mut self, verbosity: Verbosity) -> &mut Self {
+        match verbosity {
+            Verbosity::Verbose(2..) => self.args(["--build-arg", "VERBOSE=1"]),
+            _ => self,
+        }
+    }
+
+    fn scan_suggest(&mut self, suggest: bool) -> &mut Self {
+        let value = match suggest {
+            true => "true",
+            false => "false",
+        };
+        self.env("DOCKER_SCAN_SUGGEST", value)
+    }
```

Comment:
when would you need this to be false?

Reply:
We use it false already in custom images, so I decided to keep that behavior.

Reply:
my bad! I meant when would you want this to be `true`

Reply:
Ah probably never, we could just have `disable_scan_suggest` I guess?

Reply:
sounds good!